docs/src/userguide/language_basics.rst

```diff
@@ -44,72 +47,137 @@ the use of ‘early binding’ programming techniques.
 C variable and type definitions
 ===============================
 
-The :keyword:`cdef` statement is used to declare C variables, either local or
-module-level::
+C variables can be declared by using the :keyword:`cdef` statement, by annotating
+variable by special cython type or using function ``declare()``. Statement :keyword:`cdef`
+and function ``declare()`` can declare either local or
+module-level variables, but annotated version currently supports only local variables:
 
-    cdef int i, j, k
-    cdef float f, g[42], *h
+.. tabs::
 
-and C :keyword:`struct`, :keyword:`union` or :keyword:`enum` types:
+    .. group-tab:: Pure Python
 
-.. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.pyx
+        .. code-block:: python
 
-See also :ref:`struct-union-enum-styles`
+            global_x = declare(cython.int)
+            def main():
+                i: cython.int
+                j: cython.int
+                k: cython.int
+                f: cython.float
+                g: cython.int[42]
+                h: cython.p_float
 
-.. note::
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            cdef int i, j, k
```

Comment:
Still misaligned with pure python version

Reply:
The code is not 100% aligned everywhere. I have focused mainly finish the text around the code tu support both versions.

Reply:
I like how this shows off one of the main advantages of the Cython syntax: it's very concise.